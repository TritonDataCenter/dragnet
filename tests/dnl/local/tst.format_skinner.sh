#!/bin/bash

#
# tst.format_skinner.sh: tests "scan", "index", and "query" operations on
# skinner-style points as input data.
#

set -o errexit
. $(dirname $0)/../common.sh

function trace
{
	echo "#" "$@"
	"$@"
}

tmpfile="/var/tmp/$(basename $0).$$"
tmpfile2="$tmpfile.2"
echo "using tmpfiles \"$tmpfile\" and \"$tmpfile2\"" >&2

# Test points with no fields
dnl scan-file --points $DN_DATADIR/2014/05-01/one.log > $tmpfile
cat $tmpfile | trace dnl scan-file --data-format=json-skinner /dev/stdin
cat $tmpfile | trace dnl scan-file --data-format=json-skinner /dev/stdin
cat $tmpfile $tmpfile | trace dnl scan-file --data-format=json-skinner /dev/stdin
cat $tmpfile $tmpfile $tmpfile | \
    trace dnl scan-file --data-format=json-skinner /dev/stdin

# Test points with a couple of fields
dnl scan-file --points -b req.method,res.statusCode \
    $DN_DATADIR/2014/05-01/one.log > $tmpfile
dnl scan-file -b req.method $DN_DATADIR/2014/05-01/one.log
cat $tmpfile $tmpfile $tmpfile | \
    trace dnl scan-file --data-format=json-skinner /dev/stdin
cat $tmpfile $tmpfile $tmpfile | \
    trace dnl scan-file -b req.method --data-format=json-skinner /dev/stdin

# Test indexes
echo "building index"
cat $tmpfile $tmpfile $tmpfile | \
    dnl index-file -c req.method --data-format=json-skinner /dev/stdin $tmpfile2
dnl query-file $tmpfile2
dnl query-file -b req.method $tmpfile2

rm -f $tmpfile $tmpfile2
