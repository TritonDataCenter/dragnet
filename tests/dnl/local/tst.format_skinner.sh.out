# dnl scan-file --data-format=json-skinner /dev/stdin
VALUE
  250
# dnl scan-file --data-format=json-skinner /dev/stdin
VALUE
  250
# dnl scan-file --data-format=json-skinner /dev/stdin
VALUE
  500
# dnl scan-file --data-format=json-skinner /dev/stdin
VALUE
  750
REQ.METHOD VALUE
DELETE        74
GET           54
HEAD          61
PUT           61
# dnl scan-file --data-format=json-skinner /dev/stdin
VALUE
  750
# dnl scan-file -b req.method --data-format=json-skinner /dev/stdin
REQ.METHOD VALUE
DELETE       222
GET          162
HEAD         183
PUT          183
building index
VALUE
  750
REQ.METHOD VALUE
DELETE       222
GET          162
HEAD         183
PUT          183
